src/checker.py

```diff
@@ -238,9 +239,16 @@ def _collect_source_data(
         manifest_datas = self._external_data.setdefault(source_path, [])
         if any(d.source is source for d in manifest_datas):
             return
-        data = ExternalData.from_source(source_path, source)
-        if data:
-            manifest_datas.append(data)
+        try:
+            data = ExternalData.from_source(source_path, source)
+        except ManifestUnsupported as err:
+            log.debug(err)
+            return
+        except ManifestLoadError as err:
+            self._errors.append(err)
+            log.error(err)
+            return
+        manifest_datas.append(data)
```

Comment:
I am personally a fan of this syntax but I know it's an acquired taste.

```suggestion
        try:
            data = ExternalData.from_source(source_path, source)
        except ManifestUnsupported as err:
            log.debug(err)
        except ManifestLoadError as err:
            self._errors.append(err)
            log.error(err)
        else:
            manifest_datas.append(data)
```

Reply:
I usually try to keep next-level-indented blocks as short as possible, or avoid them altogether. Here `except` blocks are 1 or 2 lines, so I guess it makes sense to avoid `return`s.